internal/schemautil/userconfig/apiconvert/fromapi.go

```diff
@@ -0,0 +1,225 @@
+package apiconvert
+
+import (
+	"fmt"
+
+	"github.com/aiven/terraform-provider-aiven/internal/schemautil/userconfig"
+)
+
+// sliceHasNestedProperties is a function that checks if the given slice has nested properties.
+func sliceHasNestedProperties(vr interface{}, va map[string]interface{}) (map[string]interface{}, bool) {
+	var res map[string]interface{}
+
+	// rok is the resulting ok value.
+	var rok bool
+
+	vra, ok := vr.([]interface{})
+	if !ok {
+		return res, rok
+	}
+
+	for _, v := range vra {
+		if p, ok := v.(map[string]interface{}); ok && p != nil {
+			rok = true
+
+			break
+		}
+	}
+
+	if i, ok := va["items"].(map[string]interface{}); ok && rok {
+		if p, ok := i["properties"].(map[string]interface{}); ok {
+			res = p
+		}
+	}
+
+	if rok && len(res) == 0 {
+		rok = false
+	}
+
+	return res, rok
+}
+
+// unsettedAPIValue is a function that returns an unsetted value with the given type.
+func unsettedAPIValue(t string) interface{} {
+	var res interface{}
+
+	switch t {
+	// TODO: Uncomment when we use the actual types in the schema.
+	//case "boolean":
+	//	res = false
+	//case "integer":
+	//	res = 0
+	//case "number":
+	//	res = float64(0)
+	//case "string":
+	//	res = ""
```

Comment:
1. How do we know there is "actual types in the schema"?
2. Can it be uncommented now?

Reply:
this is planned for v4